src/responder/secrets/local.c

```diff
@@ -373,13 +373,41 @@ int local_db_delete(TALLOC_CTX *mem_ctx,
                     const char *req_path)
 {
     struct ldb_dn *dn;
+    static const char *attrs[] = { NULL };
+    struct ldb_result *res = NULL;
     int ret;
 
     ret = local_db_dn(mem_ctx, lctx->ldb, req_path, &dn);
     if (ret != EOK) return ret;
 
-    ret = ldb_delete(lctx->ldb, dn);
-    return sysdb_error_to_errno(ret);
+    ret = ldb_search(lctx->ldb, mem_ctx, &res, dn, LDB_SCOPE_BASE,
+                    attrs, LOCAL_CONTAINER_FILTER);
+    if (ret != EOK) return ret;
+
+    if (res->count == 1) {
+        talloc_free(res);
+        res = NULL;
+
+        ret = ldb_search(lctx->ldb, mem_ctx, &res, dn, LDB_SCOPE_ONELEVEL,
+                         attrs, NULL);
+        if (ret != EOK) return ret;
+
+        if (res->count > 0) {
+            ret = EINVAL;
+            DEBUG(SSSDBG_OP_FAILURE,
+                  "Failed to remove '%s': Container is not empty\n",
+                  ldb_dn_get_linearized(dn));
+
+            goto done;
+        }
+    }
+
+    ret = sysdb_error_to_errno(ldb_delete(lctx->ldb, dn));
```

Comment:
please split in two lines
'''
ret = ldb_delete(lctx->ldb, dn);
if (ret) {
    ret = sysdb_error_to_errno(ret);
}
'''